test/char.jl

```diff
@@ -348,8 +348,22 @@ end
     @test all(Base.is_overlong_enc, overlong_uints)
     @test all(Base.isoverlong, overlong_chars)
     @test all(Base.ismalformed, overlong_chars)
+    @test all(!hascodepoint, overlong_chars)
     @test repr("text/plain", overlong_chars[1]) ==
         "'\\xc0': Malformed UTF-8 (category Ma: Malformed, bad data)"
+
+    let c = '\xf0\x8e\x80\x80' # overlong but not malformed
+        @test Base.isoverlong(c)
```

Comment:
```suggestion
        @test Base.isoverlong(c)
        @test !Base.isvalid(c)
```

According to https://stackoverflow.com/questions/7113117/what-is-exactly-an-overlong-form-encoding#7113150, this must also be an invalid `Char`.